pkg/homedir/homedir.go

```diff
@@ -50,3 +36,47 @@ func GetCacheHome() (string, error) {
 	}
 	return filepath.Join(home, ".cache"), nil
 }
+
+// SetXdgDirs ensures the XDG_RUNTIME_DIR env and XDG_CONFIG_HOME variables are set.
+// containers/image uses XDG_RUNTIME_DIR to locate the auth file, XDG_CONFIG_HOME is
+// use for the containers.conf configuration file.
+func SetXdgDirs(rootless bool, rootlessUID int) error {
```

Comment:
it is up to the caller to define it.  For Podman it is `euid != 0 || !CAP_SYS_ADMIN`.

Should we hide the `rootless bool, rootlessUID int` details from this function and GetRuntimeDirUser?

Reply:
> it is up to the caller to define it.

(And how would the caller do that, without any documentation what the parameter means?)

---

Removing the parameter and delegating to `unshare.IsRootless` is not 100% an answer, but it does get us a single consistent value, so that’s 95% there.